src/community/json-ld/src/main/java/org/geoserver/jsonld/builders/impl/DynamicValueBuilder.java

```diff
@@ -96,57 +112,48 @@ private Object evaluateExpressions(JsonBuilderContext context) {
                 context = context.getParent();
                 i++;
             }
-            if (namespaces != null) prepareXpathFilter(cql);
             result = cql.evaluate(context.getCurrentObj());
         } catch (Exception e) {
             LOGGER.log(Level.INFO, "Unable to evaluate expression. Exception: {0}", e.getMessage());
         }
         return result;
     }
 
-    private void prepareXpathFilter(Expression expr) {
-        List<Expression> params = ((Function) expr).getParameters();
-        if (params.size() > 0) {
-            for (Expression e : params) {
-                if (e instanceof XPathFunction) ((XPathFunction) e).setNamespaces(namespaces);
-                else if (e instanceof Function) {
-                    prepareXpathFilter(e);
+    private void literalXpathToPropertyName(Expression expr, String literalXpath) {
```

Comment:
This function would deserve some comments explaining its behavior, and ideally should be break into small chunks. No need to change, just sharing my thoughts.

Reply:
Thanks for the review @nmco this feedback has been applied